utils/src/activityLog/Activity.ts

```diff
@@ -96,4 +102,16 @@ export abstract class ActivityBase<R> implements hTypes.Activity {
                 return this.initialState();
         }
     }
+
+    private startTimer(startTimeMs: number): void {
+        this.timer = setInterval(() => {
+            this.timerMessage = dateTimeUtils.getFormattedDurationInMinutesAndSeconds(Date.now() - startTimeMs);
+            this.report();
+        }, 1000);
+    }
+
+    public dispose(): void {
+        clearInterval(this.timer);
+        this.cancellationTokenSource.dispose();
```

Comment:
CTS is optional, right?

```suggestion
        this.cancellationTokenSource?.dispose();
```

Reply:
Actually, this class isn't the owner of this CTS, so we shouldn't be disposing it.

Reply:
Nevermind this comment chain, CTS is always defined. It's optional on the interface but always present in this class.